File: comptoiristeflutter/lib/services/api_comptoiriste.dart
import 'dart:convert';
import 'package:http/http.dart' as http;
import 'package:flutter_secure_storage/flutter_secure_storage.dart';

class ApiComptoiriste {
  static const String baseUrl = 'http://192.168.108.153:3000/api/comptoiriste';
  final _storage = const FlutterSecureStorage();

  Future<String?> _getToken() async {
    return await _storage.read(key: 'token');
  }

  Future<Map<String, String>> _getHeaders() async {
    final token = await _getToken();
    return {
      'Content-Type': 'application/json',
      'Authorization': 'Bearer $token',
    };
  }

  Future<dynamic> loginComptoiriste(String email, String password) async {
    try {
      final response = await http.post(
        Uri.parse('$baseUrl/login'),
        body: json.encode({'email': email, 'password': password}),
        headers: {'Content-Type': 'application/json'},
      );

      if (response.statusCode == 200) {
        final data = json.decode(response.body);
        await _storage.write(key: 'token', value: data['token']);
        await _storage.write(key: 'user', value: json.encode(data['user']));
        return data;
      } else {
        throw json.decode(response.body)['error'] ?? 'Login failed';
      }
    } catch (e) {
      throw 'Login error: $e';
    }
  }

  Future<List<dynamic>> fetchOrders() async {
    try {
      final response = await http.get(
        Uri.parse('$baseUrl/orders'),
        headers: await _getHeaders(),
      );

      if (response.statusCode == 200) {
        final data = json.decode(response.body);
        return data['orders'];
      } else {
        throw json.decode(response.body)['error'] ?? 'Failed to load orders';
      }
    } catch (e) {
      throw 'Orders fetch error: $e';
    }
  }

  Future<dynamic> updateOrderStatus(String orderId, String status) async {
    try {
      final response = await http.put(
        Uri.parse('$baseUrl/orders/$orderId/status'),
        body: json.encode({'status': status}),
        headers: await _getHeaders(),
      );

      if (response.statusCode == 200) {
        return json.decode(response.body);
      } else {
        throw json.decode(response.body)['error'] ?? 'Update failed';
      }
    } catch (e) {
      throw 'Update error: $e';
    }
  }

  Future<List<dynamic>> fetchRecentClients() async {
    try {
      final response = await http.get(
        Uri.parse('$baseUrl/clients/recent'),
        headers: await _getHeaders(),
      );

      if (response.statusCode == 200) {
        final data = json.decode(response.body);
        return data['clients'];
      } else {
        throw json.decode(response.body)['error'] ?? 'Failed to load clients';
      }
    } catch (e) {
      throw 'Clients fetch error: $e';
    }
  }

  Future<Map<String, dynamic>> fetchOrdersWithStats() async {
    try {
      final response = await http.get(
        Uri.parse('$baseUrl/orders'),
        headers: await _getHeaders(),
      );

      if (response.statusCode == 200) {
        return json.decode(response.body);
      } else {
        throw json.decode(response.body)['error'] ?? 'Failed to load orders with stats';
      }
    } catch (e) {
      throw 'Orders with stats fetch error: $e';
    }
  }
}

File: khiwacmp/src/services/api_comptoiriste.js
// services/api_comptoiriste.js

import axios from 'axios';

const API_URL = 'http://localhost:3000/api/comptoiriste';
const api = axios.create({
  baseURL: API_URL,
  timeout: 10000,
});
api.interceptors.request.use(config => {
  const token = localStorage.getItem('token');
  if (token) {
    config.headers.Authorization = `Bearer ${token}`;
  }
  return config;
});
export const loginComptoiriste = async (email, password) => {
    try {
      const response = await axios.post(`${API_URL}/login`, { email, password });
  
      // Stocker le token et l'utilisateur dans le localStorage
      localStorage.setItem('token', response.data.token);
      localStorage.setItem('user', JSON.stringify(response.data.user));
  
      return response.data; // Retourner les données pour une utilisation dans le composant React
    } catch (error) {
      throw error.response?.data?.error || 'Erreur de connexion';
    }
  };

// Récupérer les détails du salon


// Récupérer toutes les commandes du salon
export const fetchOrders = async () => {
  try {
    const response = await api.get('/orders');
    return response.data.orders;
  } catch (error) {
    throw error.response?.data?.error || 'Erreur lors du chargement des commandes';
  }
};

export const updateOrderStatus = async (orderId, status) => {
  try {
    const response = await api.put(`/orders/${orderId}/status`, { status });
    return response.data;
  } catch (error) {
    throw error.response?.data?.error || 'Erreur lors de la mise à jour';
  }
};

export const fetchRecentClients = async () => {
  try {
    const response = await api.get('/clients/recent');
    return response.data.clients;
  } catch (error) {
    throw error.response?.data?.error || 'Erreur lors du chargement des clients';
  }
};

export const fetchSalon = async () => {
  try {
    const response = await api.get('/salon');
    return response.data.salon;
  } catch (error) {
    throw error.response?.data?.error || 'Erreur lors du chargement du salon';
  }
};

export const fetchOrdersWithStats = async () => {
  try {
    const response = await api.get('/orders');
    return {
      orders: response.data.orders,
      stats: response.data.stats,
      dailyOrders: response.data.dailyOrders
    };
  } catch (error) {
    throw error.response?.data?.error || 'Erreur lors du chargement des commandes';
  }
};

export const fetchRecentClientsWithOrders = async () => {
  try {
    const response = await api.get('/clients/recent');
    return response.data.clients;
  } catch (error) {
    throw error.response?.data?.error || 'Erreur lors du chargement des clients';
  }
};
export const fetchIngredients = async () => {
  try {
    const response = await api.get('/ingredients');
    return response.data.ingredients;
  } catch (error) {
    throw error.response?.data?.error || 'Erreur lors du chargement des ingrédients';
  }
};

// Nouvelle fonction pour mettre à jour les stocks
// services/api_comptoiriste.js
export const updateMultipleIngredients = async (orderId, usedIngredients) => {
  try {
    const response = await api.post('/update-stock', { 
      orderId, // Ajouter orderId dans le corps
      usedIngredients 
    });
    return response.data;
  } catch (error) {
    console.error('Error updating ingredients:', error.response?.data || error);
    throw error.response?.data?.error || 'Erreur lors de la mise à jour des stocks';
  }
};
export const fetchOrderDetails = async (orderId) => {
  try {
    const response = await api.get(`/orders/${orderId}`);
    return response.data.order;
  } catch (error) {
    throw error.response?.data?.error || 'Commande non trouvée';
  }
};
// Fonction pour mettre à jour plusieurs ingrédients

// Ajoutez cette méthode pour récupérer les commandes annulées si besoin
export const fetchCancelledOrders = async () => {
  try {
    const response = await api.get('/orders?cancelled=true');
    return response.data.orders;
  } catch (error) {
    throw error.response?.data?.error || 'Erreur lors du chargement des commandes annulées';
  }
};
export default {
  loginComptoiriste,
  fetchOrders,
  updateOrderStatus,
  fetchRecentClients,
  fetchSalon,
  fetchRecentClientsWithOrders,
  fetchOrdersWithStats,
  fetchIngredients,
  fetchOrderDetails,
  
  updateMultipleIngredients
};



